src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -35,8 +35,8 @@
 
 #ifdef __GNUC__
 
-// __nop needs volatile so that compiler doesn't optimize it away
-#define NOP() asm volatile ("nop");
+// ISO C++ asm is always implicitly volatile
```

Comment:
Is it? I think this comment needs a reference, or (maybe better) it should be removed.

Reply:
I've added the reference, I don't think removing the comment explaining why the asm is declared this way is a good idea